src/components/Feature/Feature.tsx

```diff
@@ -113,29 +113,31 @@ const Feature: FC<FeatureProps> = ({ featureKey, children, fallback }) => {
           },
         }
       )
-        .then((res) => res.json())
+        .then((res) => res.json() as FeatureToggleDto)
         .catch((error) => {
           throw new Error(error);
         }),
     { enabled: portalToken !== undefined }
   );
 
   useEffect(() => {
-    const feature = featureToggle?.features?.find(
-      (feature) => feature.featureKey === featureKey
-    );
-    if (feature) {
-      if (isUserInActiveUserArray(username, feature.activeUsers)) {
-        setShowContent(true);
-      } else if (!feature.activeEnvironments?.includes(environment)) {
-        setShowContent(false);
+    if (featureToggle && featureToggle.features) {
+      const feature = featureToggle.features.find(
+        (feature) => feature.featureKey === featureKey
+      );
+      if (feature) {
```

Comment:
Could this maybe be written to check for the 'bad'-path cases instead and the return?